syntax = "proto3";
import "google/protobuf/any.proto";
package MyGame;

enum MsgType
{
    MsgType_Default = 0;
    MsgType_Login = 1;
    MsgType_Logout = 2;
    MsgType_Act = 3;
}

enum MsgErrCode
{
    MsgErr_OK = 0;
    MsgErr_Fail = 1;
    MsgErr_NotExist = 2;
    MsgErr_PasswdWrong = 3;
}

message MsgHead
{
    MsgType type = 1;
}

message MsgRspHead
{
    MsgType type = 1;
    MsgErrCode res = 2;
}

message RequestLogin
{
    string strName = 1;
    string strPass = 2;
}

message ResponseLogin
{
    uint64 roleId = 1;
    int32 roleLevel = 2;
}

message RequestLogout
{
    uint64 roleId = 1;
}

message ResponseLogout
{
    uint64 roleId = 1;
}

message RequestAct
{
    string action = 1;
}

message ResponseAct
{
    string action = 1;
}

message Msg
{
    MsgHead head = 1;
    google.protobuf.Any payload = 2;
}

message MsgRsp
{
    MsgRspHead head = 1;
    google.protobuf.Any payload = 2;
}
